src/main/java/org/hawkular/client/android/activity/MetricDataActivity.java

```diff
@@ -0,0 +1,167 @@
+import lecho.lib.hellocharts.model.PointValue;
+import lecho.lib.hellocharts.model.Viewport;
+import lecho.lib.hellocharts.view.LineChartView;
+import timber.log.Timber;
+
+public class MetricDataActivity extends AppCompatActivity {
+    @InjectView(R.id.toolbar)
+    Toolbar toolbar;
+
+    @InjectView(R.id.chart)
+    LineChartView chart;
+
+    @Override
+    protected void onCreate(Bundle state) {
+        super.onCreate(state);
+        setContentView(R.layout.activity_chart);
+
+        setUpBindings();
+
+        setUpToolbar();
+
+        setUpMetricData();
+    }
+
+    private void setUpBindings() {
+        ButterKnife.inject(this);
+    }
+
+    private void setUpToolbar() {
+        setSupportActionBar(toolbar);
+
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+    }
+
+    private void setUpMetricData() {
+        showProgress();
+
+        BackendClient.getInstance().getMetricData(getTenant(), getMetric(), this, new MetricDataCallback());
+    }
+
+    private void showProgress() {
+        ViewDirector.of(this).using(R.id.animator).show(R.id.progress);
+    }
+
+    private Tenant getTenant() {
+        return getIntent().getParcelableExtra(Intents.Extras.TENANT);
+    }
+
+    private Metric getMetric() {
+        return getIntent().getParcelableExtra(Intents.Extras.METRIC);
+    }
+
+    private void setUpMetricData(List<MetricData> metricDataList) {
+        List<PointValue> chartPoints = new ArrayList<>();
+
+        List<AxisValue> chartAxisPoints = new ArrayList<>();
+
+        for (int metricDataPosition = 0; metricDataPosition < metricDataList.size(); metricDataPosition++) {
+            MetricData metricData = metricDataList.get(metricDataPosition);
+
+            chartPoints.add(new PointValue(metricDataPosition, metricData.getValue()));
+
+            chartAxisPoints.add(new AxisValue(metricDataPosition)
+                .setLabel(DateFormat.getTimeInstance(DateFormat.SHORT).format(new Date(metricData.getTimestamp()))));
+        }
+
+        Line chartLine = new Line(chartPoints)
+            .setColor(getResources().getColor(R.color.background_primary_dark))
+            .setCubic(true)
+            .setHasPoints(false);
+
+        LineChartData chartData = new LineChartData()
+            .setLines(Collections.singletonList(chartLine));
+        chartData.setAxisXBottom(new Axis()
+            .setValues(chartAxisPoints));
+        chartData.setAxisYLeft(new Axis()
+            .setHasLines(true));
+
+        chart.setLineChartData(chartData);
+
+        Viewport chartViewport = new Viewport(chart.getMaximumViewport());
+
+        chartViewport.bottom = chart.getMaximumViewport().bottom - 50;
+        chartViewport.top = chart.getMaximumViewport().top + 50;
+
+        chart.setMaximumViewport(chartViewport);
+        chart.setCurrentViewport(chartViewport);
+
+        hideProgress();
+    }
+
+    private void hideProgress() {
+        ViewDirector.of(this).using(R.id.animator).show(R.id.chart);
```

Comment:
Oh, I thought hideProgress hides some spinner -- with that impl, perhaps change the name to `showChart()`?